File: api-main/api-main/src/Domain/Entities/InstallmentPaymentAggregate/InstallmentPayment.cs
using Domain.Common;
using Domain.Entities.OrderAggregate;

namespace Domain.Entities.InstallmentPaymentAggregate;

public class InstallmentPayment : AuditableEntity
{
    public InstallmentPayment(double amount, DateOnly date, string? description = null)
    {
        Amount = amount;
        Date = date;
    }

    public double Amount { get; set; }
    public DateOnly Date { get; set; }
    public string? Description { get; set; }
    public int OrderId { get; set; }

    public Order? Order { get; set; }
}

File: app-dev/app-dev/lib/payment/domain/installment_payment/installment_payment.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:team/order/domain/order/order.dart';

part 'installment_payment.freezed.dart';
part 'installment_payment.g.dart';

@freezed
class InstallmentPayment with _$InstallmentPayment {
  const factory InstallmentPayment({
    required final int id,
    required final int orderId,
    required final String customerFullName,
    final int? orderListId,
    required final double amount,
    required final DateTime date,
    required final DateTime lastUpdatedAt,
    // required final Product product,
  }) = _InstallmentPayment;

  factory InstallmentPayment.fromJson(Map<String, dynamic> json) => _$InstallmentPaymentFromJson(json);
}
